File: MinimalClean/src/MinimalClean.Architecture.Web/Domain/Users/Enums/UserRole.cs
using Ardalis.SmartEnum;

namespace MinimalClean.Architecture.Web.Domain.Users.Enums;

public sealed class UserRole : SmartEnum<UserRole>
{
    public static readonly UserRole Administrator = new(nameof(Administrator), 0);
    public static readonly UserRole InspectionManager = new(nameof(InspectionManager), 1);
    public static readonly UserRole Inspector = new(nameof(Inspector), 2);
    public static readonly UserRole MaintenanceEngineer = new(nameof(MaintenanceEngineer), 3);
    public static readonly UserRole Viewer = new(nameof(Viewer), 4);

    private UserRole(string name, int value) : base(name, value) { }
}

File: src/main/java/com/smartdroneinspection/users/domain/enums/UserRole.java
package com.smartdroneinspection.users.domain.enums;

import com.smartdroneinspection.shared.auth.Roles;

public enum UserRole {
  ADMIN(Roles.ADMIN),
  CLIENT(Roles.CLIENT),
  SERVICE_MANAGER(Roles.SERVICE_MANAGER),
  INSPECTOR(Roles.INSPECTOR),
  MAINTENANCE_ENGINEER(Roles.MAINTENANCE_ENGINEER);

  private final String value;

  UserRole(String value) {
    this.value = value;
  }

  public String value() {
    return value;
  }

  public boolean serviceRole() {
    return this == SERVICE_MANAGER || this == INSPECTOR || this == MAINTENANCE_ENGINEER;
  }
}
